pkg/util/detector/detector.go

```diff
@@ -528,15 +552,62 @@ func (d *ResourceDetector) BuildClusterResourceBinding(object *unstructured.Unst
 		},
 		Spec: workv1alpha1.ResourceBindingSpec{
 			Resource: workv1alpha1.ObjectReference{
-				APIVersion:      object.GetAPIVersion(),
-				Kind:            object.GetKind(),
-				Name:            object.GetName(),
-				ResourceVersion: object.GetResourceVersion(),
+				APIVersion:                  object.GetAPIVersion(),
+				Kind:                        object.GetKind(),
+				Name:                        object.GetName(),
+				ResourceVersion:             object.GetResourceVersion(),
+				ReplicaResourceRequirements: replicaResourceRequirements,
+				Replicas:                    replicas,
 			},
 		},
 	}
 
-	return binding
+	return binding, nil
+}
+
+// GetObjectResource get the replicas and resources of a object
+func (d *ResourceDetector) GetObjectResource(object *unstructured.Unstructured) (corev1.ResourceList, int32, error) {
```

Comment:
```suggestion
func (d *ResourceDetector) GetReplicaDeclaration(object *unstructured.Unstructured) (corev1.ResourceList, int32, error) {
```
How about change to this name?

Reply:
ok